staging/src/k8s.io/client-go/discovery/cached/disk/cached_discovery.go

```diff
@@ -297,7 +301,7 @@ func NewCachedDiscoveryClientForConfig(config *restclient.Config, discoveryCache
 		return nil, err
 	}
 
-	return newCachedDiscoveryClient(discoveryClient, discoveryCacheDir, ttl), nil
+	return newCachedDiscoveryClient(memory.NewMemCacheClient(discoveryClient), discoveryCacheDir, ttl), nil
```

Comment:
What is the reason of this wrapping?

Reply:
When the server returns the aggregated discovery format, it returns both the groups *and* the resources. The `ServerGroupsAndResources` (as well as other functions) is implemented by initially calling `ServerGroups`. But `ServerGroups` does not return (or store) the resources. This memcache stores the resources from the `ServerGroups` call, so there will be no need for subsequent network calls for resources.